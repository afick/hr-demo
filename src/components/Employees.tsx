
import { Department, Employee } from 'API'
import { Query } from 'class/Query'
import React, { useEffect, useState } from 'react'
import EmployeeCards from './element/EmployeeCards'

const Employees = (props: any) => {
    const hash: string = props.location.hash.replace('#', '')
    const [employees, setEmployees] = useState<Employee[] | any>([])
    useEffect(() => {
        if (props.location.state !== undefined) {
            new Query().getDepartment((props.location.state.id)).then((departments: any) => {
                setEmployees(departments.employees.items)
            })
        }
        else {
            //would refactor this with a better graphQL query
            new Query().getDepartments().then((departments: any) => {
                const dept = departments.filter((item: Department) => item.DepartmentName === hash)
                new Query().getDepartment((dept.id)).then((departments: any) => {
                    setEmployees(departments.employees.items)
                })
            })
        }

    }, [hash, props.location.state])
    return (
        <div style={{ height: '100%', width: '100%', margin: '20px' }}>
            <h1>Department: {hash}</h1>
            <EmployeeCards employees={employees} />
        </div>
    )
}


export default Employees
