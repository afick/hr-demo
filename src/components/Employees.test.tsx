import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import Employees from './Employees'
import renderer from 'react-test-renderer'


describe('Employees tests', () => {
    it('renders without crashing', () => {
        <BrowserRouter>
            const div = document.createElement('div')
            ReactDOM.render(<Employees />, div)
            ReactDOM.unmountComponentAtNode(div)
        </BrowserRouter>
    })
    it('renders snapshot as expected', () => {
        const state = { state: { id: 1 }, hash: '#test' }
        const tree = renderer.create(<Employees location={state} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

})
