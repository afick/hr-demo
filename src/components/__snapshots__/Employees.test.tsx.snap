// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees tests renders snapshot as expected 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "margin": "20px",
      "width": "100%",
    }
  }
>
  <h1>
    Department: 
    test
  </h1>
  <div>
    No records found
  </div>
</div>
`;
